api/desecapi/views/mfa.py

```diff
@@ -22,7 +22,12 @@ class TOTPViewSet(IdempotentDestroyMixin, viewsets.ModelViewSet):
     def permission_classes(self):
         if self.action == "verify" and self.request.method == "POST":
             return [AllowAny]  # temporary for anonymous activation
-        return [IsAuthenticated, permissions.HasManageTokensPermission]
+        ret = [IsAuthenticated, permissions.HasManageTokensPermission]
+        if self.request.method not in SAFE_METHODS and self.action != "verify":
```

Comment:
why (here and above) check for `action` and `method`? verify is defined for POST only, anyways.

Reply:
Once we revert the commit that allows anonymous TOTP factor activation (i.e. when we have the activaction based on double auth action, like for FIDO), the `verify` action must not require `MFARequiredIfEnabled`, because otherwise you would need a stepped-up token in order to step up. So the `verify` action is exempt here, and requires `IsLoginToken` only. Currently, this is immaterial because the `verify` action is open to all, but I'd prefer to leave it so we don't forget. (OTOH, tests cover it.)

You are right about the check above, where the method is irrelevant. I removed it.